courses/utils.py

```diff
@@ -63,7 +126,9 @@ def generate_program_certificate(user, program):
         .count()
     )
 
-    if len(courses_in_program_ids) > num_courses_with_cert:
+    if (
+        len(courses_in_program_ids) - not_required_courses_count
+    ) > num_courses_with_cert or not has_earned_program_cert(user, program):
         return None, False
```

Comment:
I'm a bit confused by this, but I don't think you need the course count stuff anymore, it should be purely based on `has_earned_program_cert()`:
```suggestion
    if not has_earned_program_cert(user, program):
        return None, False
```

Reply:
I added this because in `has_earned_program_cert()` there was a check for `courserun grade` So I added this check to verify if there are all required certificates are available. but as per your above [comment](https://github.com/mitodl/mitxonline/pull/1239#discussion_r1041439683) this not required anymore. So I removed it